### PR TITLE
Fix issue with highlighting custom tag when attribute selector presented

### PR DESCRIPTION
**Pull request by ptsurko**
Fix issue with highlighting custom tag with attribute selector

screen shot 2016-08-17 at 4 42 12 pm

PS. Added tests to cover custom tag highlighting

### CODE REVIEW

**Comment by braver** [comment]
The regex looks a lot better/smarter than the original, so if the tests pass it lgtm

**Comment by esdoppio** [comment]
See the definition of a [valid custom element name](https://www.w3.org/TR/custom-elements/#valid-custom-element-name):

* starts with [a-z]
* does not contain [A-Z]
* contains at least one hyphen for forward compatibility

I think the regexp should at least imposes these three requirements. Here's an example:

`[a-z][\w&&[^A-Z]]*+-[\w-&&[^A-Z]]++`

It's not 100% standard-compliant (like, `.` is allowed in the name, but in CSS, it must be escaped to use in an identifier, makes things complicated, let's not worry about it now), but closer than the current version.

**Comment by esdoppio** [comment]
I don't see any specs for pseudo classes/elements whose name contains a hyphen, considering this rule comes before the rules of these two, not sure if it's ok to remove the lookbehind.

Try change the first spec: `very-very-custom:hover` -> `very-very-custom:first-child`, if it still passes, then it should be ok.

**Comment by ptsurko** [comment]
@esdoppio Added more tests to cover `:first-child` case.

**Comment by esdoppio** [comment]
The spec title's amiss, `:first-child` is a pseudo class.

**Comment by ptsurko** [comment]
@esdoppio fixed

File: spec/grammar-spec.coffee [hunk 1]
```diff
@@ -203,6 +203,31 @@ describe "less grammar", ->
...
 
+  it 'parses custom tag with pseudo classes', ->
+    {tokens} = grammar.tokenizeLine("very-very-custom:hover::after { color: inherit }")
+    expect(tokens[0]).toEqual value: 'very-very-custom', scopes: ['source.css.less', 'keyword.control.html.custom.elements']
+    expect(tokens[1]).toEqual value: ":", scopes: ['source.css.less', 'entity.other.attribute-name.pseudo-class.css', 'punctuation.definition.entity.css']
+    expect(tokens[2]).toEqual value: "hover", scopes: ['source.css.less', 'entity.other.attribute-name.pseudo-class.css']
+    expect(tokens[3]).toEqual value: "::", scopes: ['source.css.less', 'entity.other.attribute-name.pseudo-element.css', 'punctuation.definition.entity.css']
+    expect(tokens[4]).toEqual value: "after", scopes: ['source.css.less', 'entity.other.attribute-name.pseudo-element.css']
+
+  it 'parses custom tag with pseudo classes containings dash symbol', ->
```
Comment (esdoppio) [inline spec/grammar-spec.coffee]: "containings dash symbol" -> "containing hyphens"

**Comment by esdoppio** [comment]
Would you update the regexp too? If you don't want to, I'll open a PR myself. :)

**Pull request closed without merging**